packages/yarnpkg-core/sources/Installer.ts

```diff
@@ -21,7 +21,30 @@ export type FinalizeInstallStatus = {
   buildDirective: Array<BuildDirective>,
 };
 
+export type PackageMeta = Record<any, any>;
+
 export interface Installer {
+  /**
+   * Returns cache key used to store package meta information.
```

Comment:
Needs to describe why this function matters, and what happens if it's set to fixed strings. In this kind of setup, a good idea is to explain how we use it in the PnP linker and what would happen if it was incorrectly written.